osu.Game/Overlays/Mods/ModSection.cs

```diff
@@ -127,18 +127,28 @@ public void DeselectTypes(IEnumerable<Type> modTypes, bool immediate = false)
         }
 
         /// <summary>
-        /// Select one or more mods in this section and deselects all other ones.
+        /// Updates all buttons with the given list of selected mods.
         /// </summary>
-        /// <param name="modTypes">The types of <see cref="Mod"/>s which should be selected.</param>
-        public void SelectTypes(IEnumerable<Type> modTypes)
+        /// <param name="newSelectedMods">The types of <see cref="Mod"/>s to select.</param>
+        public void UpdateSelectedMods(IReadOnlyList<Mod> newSelectedMods)
         {
             foreach (var button in buttons)
             {
-                int i = Array.FindIndex(button.Mods, m => modTypes.Any(t => t == m.GetType()));
+                int index = -1;
 
-                if (i >= 0)
-                    button.SelectAt(i);
-                else
+                foreach (var mod in newSelectedMods)
+                {
+                    index = Array.FindIndex(button.Mods, m1 => mod.GetType() == m1.GetType());
+                    if (index < 0)
+                        continue;
+
+                    var buttonMod = button.Mods[index];
+                    buttonMod.CopyFrom(mod);
+                    button.SelectAt(index);
+                    break;
+                }
+
+                if (index < 0)
```

Comment:
Not a fan of how this loop is written. I'd probably prefer something along the lines of

```csharp
        public void UpdateSelectedMods(IReadOnlyList<Mod> newSelectedMods)
        {
            foreach (var button in buttons)
                updateButtonMods(button, newSelectedMods);
        }

        private void updateButtonMods(ModButton button, IReadOnlyList<Mod> newSelectedMods)
        {
            foreach (var mod in newSelectedMods)
            {
                int index = Array.FindIndex(button.Mods, m1 => mod.GetType() == m1.GetType());

                if (index >= 0)
                    continue;

                var buttonMod = button.Mods[index];
                buttonMod.CopyFrom(mod);
                button.SelectAt(index);
                return;
            }

            button.Deselect();
        }
```

as I think it is clearer to read (no more confusion which loop is being continued/broken out of, early-return helps get rid of the weird outer index).

Reply:
sure but I believe that `index >= 0` check in the bottom method is flipped, yeah?

Reply:
that it is, yeah. i messed with that code a few times and missed that.